server/src/analyser.ts

```diff
@@ -1,207 +1,214 @@
 // tslint:disable:no-submodule-imports
-import { ASTNode, Document } from 'tree-sitter'
+import * as fs from 'fs'
+import * as glob from 'glob'
+import * as Path from 'path'
+
+import { Document } from 'tree-sitter'
 import * as bash from 'tree-sitter-bash'
-import {
-  Diagnostic,
-  DiagnosticSeverity,
-  Location,
-  Range,
-  SymbolInformation,
-  SymbolKind,
-} from 'vscode-languageserver/lib/main'
-
-// Global mapping from tree-sitter node type to vscode SymbolKind
-type Kinds = { [type: string]: SymbolKind }
-const kinds: Kinds = {
-  environment_variable_assignment: SymbolKind.Variable,
-  function_definition: SymbolKind.Function,
-}
+import * as LSP from 'vscode-languageserver'
+
+import * as TreeSitterUtil from './util/tree-sitter'
 
-// Global map of all the declarations that we've seen, indexed by file
-// and then name.
-type Declarations = { [name: string]: SymbolInformation[] }
+type Kinds = { [type: string]: LSP.SymbolKind }
+
+type Declarations = { [name: string]: LSP.SymbolInformation[] }
 type FileDeclarations = { [uri: string]: Declarations }
-const declarations: FileDeclarations = {}
 
-// Global map from uri to the tree-sitter document.
 type Documents = { [uri: string]: Document }
-const documents: Documents = {}
-
-// Global mapping from uri to the contents of the file at that uri.
-// We need this to find the word at a given point etc.
 type Texts = { [uri: string]: string }
-const texts: Texts = {}
 
 /**
- * Analyze the given document, cache the tree-sitter AST, and iterate over the
- * tree to find declarations.
- *
- * Returns all, if any, syntax errors that occurred while parsing the file.
- *
+ * The Analyzer uses the Abstract Syntax Trees (ASTs) that are provided by
+ * tree-sitter to find definitions, reference, etc.
  */
-export function analyze(uri: string, contents: string): Diagnostic[] {
-  const d = new Document()
-  d.setLanguage(bash)
-  d.setInputString(contents)
-  d.parse()
-
-  documents[uri] = d
-  declarations[uri] = {}
-  texts[uri] = contents
-
-  const problems = []
-
-  forEach(d.rootNode, n => {
-    if (n.type === 'ERROR') {
-      problems.push(
-        Diagnostic.create(
-          range(n),
-          'Failed to parse expression',
-          DiagnosticSeverity.Error,
-        ),
-      )
-      return
-    } else if (isDefinition(n)) {
-      const named = n.firstNamedChild
-      const name = contents.slice(named.startIndex, named.endIndex)
-      const namedDeclarations = declarations[uri][name] || []
-
-      namedDeclarations.push(
-        SymbolInformation.create(name, kinds[n.type], range(named), uri),
-      )
-      declarations[uri][name] = namedDeclarations
+export class Analyzer {
```

Comment:
I think we should an `export default class Analyzer` here, so import would be `import Analyzer from ...`.